### DISCUSSION TITLE
wrapper: why is `languageId` needed when creating user config?

### DISCUSSION THREAD

**Discussion by alissa-tung** [TypeFox/monaco-languageclient#638] [Q&A] [answered]
seems that `languageId` are used to set language id after create model reference. My question is why is `languageId` needed when creating user config, and can I create a editor with multiple language support, where the language mode is choice from the language selector per model?

**Comment by kaisalmen** [comment, marked as answer]
Hi @alissa-tung when you start the wrapper it creates the model internally and then needs that `languageId`, but on `modelUpdate` it seems the behaviour is wrong as the previously configured `languageId` is used and not the one contained in the model. This is too restrictive and should be enhanced. This is legacy behaviour that did not grow properly with the rest of the changes.

I opened #639 to track this. Thank you for bringing this up.

**Reply by alissa-tung** [reply]
thank you

**Reply by kaisalmen** [reply]
@alissa-tung a new major version of `monaco-editor-wrapper` is available: https://www.npmjs.com/package/monaco-editor-wrapper/v/5.0.0
